pkg/infrastructure/gcp/clusterapi/firewallrules.go

```diff
@@ -215,31 +215,42 @@ func deleteFirewallRule(ctx context.Context, name, projectID string) error {
 	return nil
 }
 
+func hasFirewallPermission(ctx context.Context, projectID string) (bool, error) {
+	client, err := gcpconfig.NewClient(context.Background())
+	if err != nil {
+		return false, fmt.Errorf("failed to create client during firewall rule creation: %w", err)
+	}
+
+	permissions, err := client.GetProjectPermissions(ctx, projectID, []string{
+		gcpFirewallPermission,
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to find project permissions during firewall creation: %w", err)
+	}
+
+	hasPermission := permissions.Has(gcpFirewallPermission)
+	if !hasPermission {
+		logrus.Warnf("failed to find permission %s, skipping firewall rule creation", gcpFirewallPermission)
+	}
+
+	return hasPermission, nil
+}
+
 // createFirewallRules creates the rules needed between the worker and master nodes.
 func createFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, network string) error {
+	projectID := in.InstallConfig.Config.Platform.GCP.ProjectID
 	if projID := in.InstallConfig.Config.GCP.NetworkProjectID; projID != "" {
```

Comment:
```suggestion
	if in.InstallConfig.Config.GCP.NetworkProjectID != "" 
	
	projectID = in.InstallConfig.Config.GCP.NetworkProjectID
```
Should be consistent with how this is done below at line 321